src/Cryptol/Eval/Concrete.hs

```diff
@@ -144,6 +141,15 @@ toExpr prims t0 v0 = findOne (go t0 v0)
       (_,VNumPoly{}) -> mzero
       _ -> mismatch
     where
+      intlit n
+        | n < 0 = EApp (EProofApp (ETApp (prim "negate") tInteger))
+                       (numlit (-n) tInteger)
+        | otherwise = numlit n tInteger
+
+      numlit i t =
+        EProofApp (ETApp (ETApp (prim "number") (tNum (i::Integer))) t)
```

Comment:
Probably worth a comment that `numlit i t` has the precondition that `i` should be a non-negative integer (hence why `intlit` has a special case for negative integers to satisfy the precondition).